cmd/bad-key-revoker/main_test.go

```diff
@@ -0,0 +1,516 @@
+package main
+
+import (
+	"context"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/letsencrypt/boulder/core"
+	corepb "github.com/letsencrypt/boulder/core/proto"
+	"github.com/letsencrypt/boulder/mocks"
+	rapb "github.com/letsencrypt/boulder/ra/proto"
+	"github.com/letsencrypt/boulder/sa"
+	"github.com/letsencrypt/boulder/test"
+	"github.com/letsencrypt/boulder/test/vars"
+	"google.golang.org/grpc"
+)
+
+func TestMain(m *testing.M) {
+	if !strings.HasSuffix(os.Getenv("BOULDER_CONFIG_DIR"), "config-next") {
+		os.Exit(0)
+	}
+	os.Exit(m.Run())
+}
+
+func TestSelectUncheckedRows(t *testing.T) {
+	dbMap, err := sa.NewDbMap(vars.DBConnSAFullPerms, 0)
+	test.AssertNotError(t, err, "failed setting up db client")
+	defer test.ResetSATestDatabase(t)()
+
+	bkr := &badKeyRevoker{dbMap: dbMap, uncheckedBatchSize: 1}
+
+	hashA := make([]byte, 32)
+	hashA[0] = 1
+	hashB := make([]byte, 32)
+	hashB[0] = 2
+	hashC := make([]byte, 32)
+	hashC[0] = 3
+	_, err = dbMap.Exec(`INSERT INTO blockedKeys
```

Comment:
I'd like to see these repeated blocks broken out into functions (possibly functions internal to the test case).